src/Workleap.Extensions.OpenAPI.Analyzers/CompareTypedResultWithAnnotationAnalyzer.cs

```diff
@@ -136,15 +136,23 @@ public void ValidateEndpointResponseWithAnnotationType(SymbolAnalysisContext con
         private void ValidateAnnotationWithTypedResult(SymbolAnalysisContext context, AttributeData attribute,
             Dictionary<int, List<ITypeSymbol>> methodSignatureStatusCodeToTypeMap)
         {
-            if (attribute.AttributeClass == null || attribute.ConstructorArguments.Length == 0)
+            if (attribute.AttributeClass == null || attribute.ConstructorArguments.Length == 0 && attribute.NamedArguments.Length == 0)
             {
                 return;
             }
 
-            // For the annotations [ProducesResponseType(<StatusCode>)] and [ProducesResponseType(<typeof()>, <StatusCode>)] 
+            // For the annotations [ProducesResponseType(<StatusCode>)], [ProducesResponseType(<typeof()>, <StatusCode>)] and [ProducesResponseType(<StatusCode>, Type = <typeof()>)] 
             if (attribute.AttributeClass.Equals(this.ProducesResponseSymbol, SymbolEqualityComparer.Default))
             {
-                if (attribute.ConstructorArguments.Length == 1)
+                if (TryGetTypeFromNamedArguments(attribute, out var typeFromNamedArgument))
+                {
+                    if (attribute.ConstructorArguments[0].Value is int statusCodeValue && typeFromNamedArgument != null)
```

Comment:
We could use this attribute https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis#conditional-post-conditions-notnullwhen-maybenullwhen-and-notnullifnotnull so compilre know it can't be null if method return true.

Reply:
this would be very neat, but I think it's incompatible given we currently use netstandard2.0; the NotNullWhen attribute was

Reply:
That's why you should add a package such as [Meziantou.Polyfill](https://www.nuget.org/packages/Meziantou.Polyfill) or [Polyfill](https://www.nuget.org/packages/Polyfill)